src/rdo.rs

```diff
@@ -853,14 +855,25 @@ pub fn rdo_partition_decision(
         //pmv = best_pred_modes[0].mvs[0];
 
         assert!(best_pred_modes.len() <= 4);
-        let bs = bsize.width_mi();
-        let hbs = bs >> 1; // Half the block size in blocks
-        let partitions = [
-          bo,
-          &BlockOffset{ x: bo.x + hbs as usize, y: bo.y },
-          &BlockOffset{ x: bo.x, y: bo.y + hbs as usize },
-          &BlockOffset{ x: bo.x + hbs as usize, y: bo.y + hbs as usize }
-        ];
+
+        let hbsw = subsize.width_mi(); // Half the block size width in blocks
```

Comment:
This could be factored out probably?

Reply:
Not sure what you meant but I derived hbsw, since it is referenced more than twice down below for p1 and p3?

Reply:
The whole block :)

Reply:
Oh, you probably mean another identical block of code shown in encode_partition_topdown(), yes I believe so, like given decide_partitions(partition) { ... }.

Reply:
If I want to factor out line 878 to 885, say get_sub_partitions(partition, subsize, bo)  -> partitions, 
I need to create (allocate) then return some (or all) of p1,p2, p3, which are pointer to BlockOffset.
What would be best way to create those values (pointers to BlockOffset) then return them that can be still live even after get_sub_partitions() function call is finished? So, with not passing p1,p2,p3 to get_sub_partitions().

Reply:
I somehow missed this comment